src/main/java/com/minecolonies/coremod/client/gui/WindowPostBox.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    /**
+     * Action executed when request is clicked.
+     * @param button the clicked button.
+     */
+    private void requestClicked(final Button button)
+    {
+        final int row = stackList.getListElementIndexByPane(button);
+        final ItemStack stack = allItems.get(row);
+        int qty = stack.getMaxStackSize();
+        for (final Pane child : button.getParent().getChildren())
+        {
+            if (child.getID().equals(INPUT_QTY))
+            {
+                try
+                {
+                    qty = Integer.parseInt(((TextField) child).getText());
+                    if (qty > stack.getMaxStackSize())
+                    {
+                        qty = stack.getMaxStackSize();
+                    }
+                }
+                catch (final NumberFormatException ex)
+                {
+                    //Be quiet about it.
+                }
+            }
+        }
+        MineColonies.getNetwork().sendToServer(new PostBoxRequestMessage(buildingView, stack, qty));
+    }
+
+    @Override
+    public void onOpened()
+    {
+        super.onOpened();
+        updateResources();
+    }
+
+    /**
+     * Update the item list.
+     */
+    private void updateResources()
+    {
+        final Predicate<ItemStack> filterPredicate = stack -> filter.isEmpty()
+                                                                || stack.getTranslationKey().toLowerCase(Locale.US).contains(filter.toLowerCase(Locale.US))
+                                                                || stack.getDisplayName().toLowerCase(Locale.US).contains(filter.toLowerCase(Locale.US));
+        allItems.clear();
+        allItems.addAll(getBlockList(filterPredicate));
+        updateResourceList();
+    }
+
+    /**
+     * Get the list of blocks which should be added.
+     *
+     * @param filterPredicate the predicate to filter all blocks for.
+     * @return an immutable list of blocks.
+     */
+    private Collection<? extends ItemStack> getBlockList(final Predicate<ItemStack> filterPredicate)
+    {
+        return ImmutableList.copyOf(StreamSupport.stream(Spliterators.spliteratorUnknownSize(Item.REGISTRY.iterator(), Spliterator.ORDERED), false).flatMap(item -> {
+            final NonNullList<ItemStack> stacks = NonNullList.create();
+            try
+            {
+                item.getSubItems(CreativeTabs.SEARCH, stacks);
+            }
+            catch (final Exception ex)
+            {
+                Log.getLogger().warn("Failed to get sub items from: " + item.getRegistryName(), ex);
+            }
+
+            return stacks.stream().filter(filterPredicate);
+        }).collect(Collectors.toList()));
+    }
+
+    /**
+     * Action when a button opening an inventory is clicked.
+     */
+    private void inventoryClicked()
+    {
+        MineColonies.getNetwork().sendToServer(new OpenInventoryMessage(buildingView.getID()));
```

Comment:
When I tested this the inventory said Chest.
I would prefer it when you also fixed that: Open building inventory then render the inventory string.
This can be done by returning a custom name.